engine/runtime/src/test/java/org/enso/example/TestClass.java

```diff
@@ -19,6 +20,32 @@ public static long add(long a, long b) {
     return a + b;
   }
 
+  public static double doubleArrayAverage(double[] arr) {
+    var sum = 0.0;
+    for (int i = 0; i < arr.length; i++) {
+      sum += arr[i];
+    }
+    return sum / arr.length;
+  }
+
+  public static double numberArrayAverage(Number[] arr) {
+    var sum = 0.0;
+    for (int i = 0; i < arr.length; i++) {
+      sum += arr[i].doubleValue();
+    }
+    return sum / arr.length;
+  }
+
+  public static String exactArrayAverage(Number[] arr) {
+    var sum = BigInteger.ZERO;
+    for (int i = 0; i < arr.length; i++) {
+      var n = arr[i] instanceof Long l ? BigInteger.valueOf(l) :
+          new BigInteger(arr[i].toString());
```

Comment:
If one needs to get the precise value of `EnsoBigInteger` one can use its `toString()` method and parse the value back.

Alternative would be to make the number implement `Callable<java.math.BigInteger>` or something other JDK interface (`Supplier`, `Function`, etc.) to provide the internal instance of `BigInteger`.